### ISSUE TITLE
Semaphores crash when linked into obs-studio-ftl on Linux

### ISSUE THREAD

**Issue by GregorR** [WatchBeam/ftl-sdk#34]
This is a very weird bug, but in fact the only thing that stands in the way of obs-studio-ftl+ftl-sdk working flawlessly on Linux.

ftl-sdk uses functions named `os_sem_post`, `os_sem_pend`, etc, to wrap semaphores, and similar functions to wrap mutexes. obs uses identically-named functions. Because of how shared object files work on Linux (and all ELF-using Unixes), and because obs for some reason I'm unaware of uses `-rdynamic` and thus exposes all of its own symbols, when libftl.so is linked into obs, the obs host's implementation of those functions will be used instead of libftl's. In the case of the mutex code this is fine since obs's implementations are identical, but the semaphore code is not. When `os_sem_post` is called, there's an internal error in pthread's `sem_post` and the whole thing crashes and burns.

The quickfix: Just use different function names. I accomplished this by adding the following block to libftl/posix/threads.h, before the relevant function prototypes:
```
/* Avoid clashing with host binary functions */
#define os_sem_create ftl_os_sem_create
#define os_sem_pend ftl_os_sem_pend
#define os_sem_post ftl_os_sem_post
#define os_sem_delete ftl_os_sem_delete
```
With that block in place, obs-studio-ftl works flawlessly on Linux.

A better solution would be to actually change the names, and a betterer solution would be in obs-ftl instead of here, being to not use the big `-rdynamic` hammer when surely something else would do. It's also quite conceivable that `-rdynamic` is a specific consequence of some odd configuration of my particular build, of course, but as I believe my build is fairly usual, surely other people will have the same issue.

Didn't post this as a pull request because (a) it's trivial and (b) the solution I've presented is sufficient but not ideal.

**Comment by GregorR** [comment]
[After a bit of digging, it turns out that CMake always enables `-rdynamic` for some insane reason, proudly cementing its place in the terrible build system hall of fame. Probably the best bet would be to change the names.]

**Comment by sslivins** [comment]
ah, interesting, thanks for doing the leg work on this issue @GregorR .  I'll rename those and get that out asap (hopefully today)

**Comment by GregorR** [comment]
Super. Happy to help. Once I got it working I was impressed by the latency (or lack thereof), so thanks for making it all open and thus giving me the opportunity to fix this in the first place :)

**Comment by sslivins** [comment]
fixed: 8a265a64d2e4dbf3ca61d1ac1f019aaed3f8ba0d
